File: Core CS/Algorithms/Implementation/Number Line Jumps/numberLineJumps.java
import java.io.*;
import java.util.*;
import java.text.*;

public class Solution {

    static String numberLineJumps(int x1, int v1, int x2, int v2) {
        if(v2 >= v1) return "NO";
        return (x2-x1)%(v1-v2) == 0? "YES":"NO";
    }

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int x1 = in.nextInt();
        int v1 = in.nextInt();
        int x2 = in.nextInt();
        int v2 = in.nextInt();
        String result = numberLineJumps(x1, v1, x2, v2);
        System.out.println(result);
    }
}

File: Core CS/Algorithms/Implementation/Number Line Jumps/numberLineJumps.kt
import java.io.*
import java.text.*
import java.util.*

fun numberLineJumps(x1: Int, v1: Int, x2: Int, v2: Int): String {
    if(v2 >= v1)
        return "NO"
    if((x2-x1)%(v1-v2) == 0) 
        return "YES"
    else
        return "NO"
}

fun main(args: Array<String>) {
    val scan = Scanner(System.`in`)
    val x1V1X2V2 = scan.nextLine().split(" ")
    val x1 = x1V1X2V2[0].trim().toInt()
    val v1 = x1V1X2V2[1].trim().toInt()
    val x2 = x1V1X2V2[2].trim().toInt()
    val v2 = x1V1X2V2[3].trim().toInt()

    val result = numberLineJumps(x1, v1, x2, v2)

    println(result)
}
